src/Compilers/CSharp/Test/Emit/CodeGen/CodeGenTupleTest.cs

```diff
@@ -11453,5 +11639,658 @@ static void Main()
+
+            var int_string1 = comp.CreateTupleTypeSymbol(ImmutableArray.Create(intType, stringType));
+            var int_object1 = comp.CreateTupleTypeSymbol(ImmutableArray.Create(intType, objectType));
+
+            var int_string2 = comp.GetWellKnownType(WellKnownType.System_ValueTuple_T2).Construct((TypeSymbol)intType, (TypeSymbol)stringType);
+            var int_object2 = comp.GetWellKnownType(WellKnownType.System_ValueTuple_T2).Construct((TypeSymbol)intType, (TypeSymbol)objectType);
+
+
+            Assert.Equal(ConversionKind.ImplicitTuple, comp.ClassifyConversion(int_string1, int_object2).Kind);
+            Assert.Equal(ConversionKind.ImplicitTuple, comp.ClassifyConversion(int_string2, int_object1).Kind);
+            Assert.Equal(ConversionKind.ImplicitTuple, comp.ClassifyConversion(int_string2, int_object2).Kind);
+
+            Assert.Equal(ConversionKind.NoConversion, comp.ClassifyConversion(int_object2, int_string1).Kind);
+            Assert.Equal(ConversionKind.NoConversion, comp.ClassifyConversion(int_object1, int_string2).Kind);
+            Assert.Equal(ConversionKind.NoConversion, comp.ClassifyConversion(int_object2, int_string2).Kind);
+        }
+
+        [Fact]
+        public void ClassifyConversionImplicit03()
+        {
+            var tupleComp1 = CreateCompilationWithMscorlib(trivial2uple + trivial3uple + trivalRemainingTuples);
+            var tupleComp2 = CreateCompilationWithMscorlib(trivial2uple + trivial3uple + trivalRemainingTuples);
+            var comp = CSharpCompilation.Create("test", references: new[] { MscorlibRef, tupleComp1.ToMetadataReference(), tupleComp2.ToMetadataReference() });
+
+            ITypeSymbol intType = comp.GetSpecialType(SpecialType.System_Int32);
+            ITypeSymbol stringType = comp.GetSpecialType(SpecialType.System_String);
+            ITypeSymbol objectType = comp.GetSpecialType(SpecialType.System_Object);
+
+            var int_string1 = tupleComp1.CreateTupleTypeSymbol(ImmutableArray.Create(intType, stringType));
+            var int_string2 = tupleComp2.CreateTupleTypeSymbol(ImmutableArray.Create(intType, stringType));
+
+            var int_object = tupleComp1.CreateTupleTypeSymbol(ImmutableArray.Create(intType, objectType));
+
+            Assert.Equal(ConversionKind.ImplicitTuple, comp.ClassifyConversion(int_string1, int_string2).Kind);
+            Assert.Equal(ConversionKind.ImplicitTuple, comp.ClassifyConversion(int_string2, int_string1).Kind);
+            Assert.Equal(ConversionKind.ImplicitTuple, comp.ClassifyConversion(int_string1, int_object).Kind);
+            Assert.Equal(ConversionKind.ImplicitTuple, comp.ClassifyConversion(int_string2, int_object).Kind);
+
+            Assert.Equal(ConversionKind.NoConversion, comp.ClassifyConversion(int_object, int_string1).Kind);
+            Assert.Equal(ConversionKind.NoConversion, comp.ClassifyConversion(int_object, int_string2).Kind);
```

Comment:
Thee are tests for conversions between tuples and underlying, including such corner cases where tuple types are not wellknown types. 
There are also tests for conversions between tuples and underlying types and just between underlying types (for example: ClassifyConversionImplicit02)

There is coverage for:
Cases that expressible in the source produce same results as they would in source.
Cases that are not expressible in the source produce consistent results as well.
I have pairwise coverage of all interesting and trivial cases. 

I think trying all possible permutations is unnecessary, since it is just hitting the same implementation again via a different entry point.